pyproject.toml

```diff
@@ -76,42 +76,58 @@ dependencies = [
 dynamic = ["version"]
 
 [project.optional-dependencies]
+docs = []
+lint = []
+test = []
+
+[[project.authors]]
+name = "Georg Brandl"
+email = "georg@python.org"
+
+[project.scripts]
+sphinx-build = "sphinx.cmd.build:main"
+sphinx-quickstart = "sphinx.cmd.quickstart:main"
+sphinx-apidoc = "sphinx.ext.apidoc:main"
+sphinx-autogen = "sphinx.ext.autosummary.generate:main"
+
+[dependency-groups]
+all = [
+    { include-group = "docs" },
+    { include-group = "lint" },
+    { include-group = "test" },
+    { include-group = "types" },
+]
```

Comment:
Open question: ought we use `all`, which seems more natural to me, or `dev`, which is special-cased by `uv`?

Reply:
You can change `tool.uv.default-groups` to `["all"]` if you want